spawn.c

```diff
@@ -412,9 +412,15 @@ spawn_child_exited(pid_t pid, int status)
  *
  * @tparam string|table cmd The command to launch.
  * @tparam[opt=true] boolean use_sn Use startup-notification?
- * @tparam[opt=false] boolean stdin Return a fd for stdin?
- * @tparam[opt=false] boolean stdout Return a fd for stdout?
- * @tparam[opt=false] boolean stderr Return a fd for stderr?
+ * @tparam[opt=false] boolean|string stdin Pass `true` to return a fd for stdin.
+ *   Use `"DEV_NULL"` to redirect to /dev/null, or `"INHERIT"` to inherit the
+ *   parent's stdin.
```

Comment:
could you also mention smth, that prior to glib version X.XX only boolean value is allowed?

Reply:
@actionless this isn't true though (I especially wanted to avoid that behavior,since it's hard to check from lua).
before glib 4.74, the branches to explicitly set the defaults are compiled out, however, _no error will occur_
This should be an implementation detail which is invisible from the outside.

Reply:
that's the point of documenting it - to not confuse the user why value is set but nothing happening

Reply:
I'll put it in the docs if you'd like me to, however, what i'm trying to say is that on _any_ version of glib, the behavior is the same, since these flags just _explicitly_ set the glib default behavior (ie, it's the same as passing `false`)

Reply:
so, would be good to mention it all here in the docstring 👍

Reply:
@actionless I've added this to the docstring.
Here's the updated (rendered) docs:
![2024-07-03_11-38](https://github.com/awesomeWM/awesome/assets/117116764/880aa977-6df2-4c70-9020-0734f5b8305c)